### PR TITLE
close hearing changes after admitting a case

### PR DESCRIPTION
**Pull request by SureshSoren**
1. During the BNSS flow => closing the scheduled hearing if the Judge Admits the case before getting the response from Respondent

## Summary by CodeRabbit

- **New Features**
	- Enhanced case handling with new status conditions for "PENDING_ADMISSION" and "PENDING_RESPONSE".
	- Dynamic updates to the modal interface based on case status, improving user experience.
	- Improved API interactions for case admission and hearing management, including structured request payloads and error handling.
	- New filtering logic in the Summons and Warrants modal based on task order type.
	- Expanded configuration for pending task order actions with new parameters.

- **Bug Fixes**
	- Robust date formatting that returns an empty string for invalid dates.

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js [hunk 1]
```diff
@@ -446,9 +446,25 @@ function CaseFileAdmission({ t, path }) {
...
+      });
+      if (caseDetails?.status === "PENDING_RESPONSE") {
+        const hearingData = HearingList?.find((list) => list?.hearingType === "ADMISSION" && list?.status === "SCHEDULED");
+        hearingData.workflow = hearingData.workflow || {};
+        hearingData.workflow.action = "ABANDON";
+        await Digit.HearingService.updateHearings(
+          { tenantId, hearing: hearingData, hearingType: "", status: "" },
+          { applicationNumber: "", cnrNumber: "" }
+        );
+      }
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js]: Addressed the changes added this condition 

hearingData = HearingList?.find((list) => list?.hearingType === "ADMISSION" && list?.status === "SCHEDULED") || {};

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js [hunk 2]
```diff
@@ -167,7 +167,7 @@ function CaseFileAdmission({ t, path }) {
           },
         },
       };
-      DRISTIService.customApiService(Urls.dristi.ordersCreate, orderBody, { tenantId })
+      DRISTIService.customApiService(Urls.dristi.ordersCreate, { order: orderBody }, { tenantId })
         .then((res) => {
           history.push(`/digit-ui/employee/orders/generate-orders?filingNumber=${caseDetails?.filingNumber}&orderNumber=${res.order.orderNumber}`, {
             caseId: caseDetails?.id,
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js:173]: addressed.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 3]
```diff
@@ -125,9 +125,14 @@ const AdmittedCases = () => {
...
-        caseData?.criteria[0]?.responseList[0]?.status
-      ),
+      [
+        "CASE_ADMITTED",
+        "ADMISSION_HEARING_SCHEDULED",
+        "PENDING_ADMISSION_HEARING",
+        "PENDING_NOTICE",
+        "PENDING_RESPONSE",
+        "PENDING_ADMISSION",
+      ].includes(caseData?.criteria[0]?.responseList[0]?.status),
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js]: Addressed.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 4]
```diff
@@ -763,9 +772,25 @@ const AdmittedCases = () => {
...
+      });
+      if (caseDetails?.status === "PENDING_RESPONSE") {
+        const hearingData = HearingList?.find((list) => list?.hearingType === "ADMISSION" && list?.status === "SCHEDULED");
+        hearingData.workflow = hearingData.workflow || {};
+        hearingData.workflow.action = "ABANDON";
+        await Digit.HearingService.updateHearings(
+          { tenantId, hearing: hearingData, hearingType: "", status: "" },
+          { applicationNumber: "", cnrNumber: "" }
+        );
+      }
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js]: Addressed.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 5]
```diff
@@ -626,10 +632,13 @@ const AdmittedCases = () => {
...
-    const month = String(date.getMonth() + 1).padStart(2, "0");
-    const year = date.getFullYear();
-    return `${day}-${month}-${year}`;
+    if (date) {
+      const day = String(date.getDate()).padStart(2, "0");
+      const month = String(date.getMonth() + 1).padStart(2, "0");
+      const year = date.getFullYear();
+      return `${day}-${month}-${year}`;
+    }
+    return "";
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js]: Addressed.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 6]
```diff
@@ -665,7 +674,7 @@ const AdmittedCases = () => {
           },
         },
       };
-      return DRISTIService.customApiService(Urls.dristi.ordersCreate, orderBody, { tenantId })
+      return DRISTIService.customApiService(Urls.dristi.ordersCreate, { order: orderBody }, { tenantId })
         .then((res) => {
           history.push(`/digit-ui/employee/orders/generate-orders?filingNumber=${caseDetails?.filingNumber}&orderNumber=${res.order.orderNumber}`, {
             caseId: caseDetails?.id,
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:682]: Addressed.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 7]
```diff
@@ -763,9 +776,25 @@ const AdmittedCases = () => {
...
+      });
+      if (caseDetails?.status === "PENDING_RESPONSE") {
+        const hearingData = HearingList?.find((list) => list?.hearingType === "ADMISSION" && list?.status === "SCHEDULED") || {};
+        hearingData.workflow = hearingData.workflow || {};
+        hearingData.workflow.action = "ABANDON";
+        await Digit.HearingService.updateHearings(
+          { tenantId, hearing: hearingData, hearingType: "", status: "" },
+          { applicationNumber: "", cnrNumber: "" }
+        );
+      }
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:797]: Addressed.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 8]
```diff
@@ -125,9 +134,7 @@ const AdmittedCases = () => {
   const showTakeAction = useMemo(
     () =>
       (userRoles.includes("JUDGE_ROLE") || userRoles.includes("BENCHCLERK_ROLE")) &&
-      ["CASE_ADMITTED", "ADMISSION_HEARING_SCHEDULED", "PENDING_ADMISSION_HEARING", "PENDING_NOTICE", "PENDING_RESPONSE"].includes(
-        caseData?.criteria[0]?.responseList[0]?.status
-      ),
+      relevantStatuses.includes(caseData?.criteria[0]?.responseList[0]?.status),
```
[the remark this answers is not in the document]
Comment (SureshSoren) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:137]: Addressed

**Review by anirudh-0 — APPROVED** [review]

**Pull request merged**